### DISCUSSION TITLE
Preview and  function before replacing

### DISCUSSION THREAD

**Discussion by luke2023** [theJayTea/WritingTools#35] [Ideas]
Thank you so much to the contributors of the project I pretty much liked it!

I am personally using a intel MacBook Pro 2019, and I find it good to have llama 3.2 1b as the LLM.
A small suggestion is that I hope we can have a preview window like what apple do can even further revise the generated text before pasting it .

Thank you again for the work!
Luke

**Comment by luke2023** [comment]
also, I made a small shortcut to activate the process! https://www.icloud.com/shortcuts/3c67063fdb334299b78ecb407d456ea1

**Comment by theJayTea** [comment]
Hi @luke2023 ! Thanks for those kind words. I'm happy to share that the latest update of Writing Tools now has a preview window for the Summarise, Key Points, and Table options :)

By the way, how did you get it running on macOS?

Did you use the version in the `Windows_and_Linux` folder?

And did it work straight out of the box with the python main.py command?

I'd love to test it myself on macOS, but my Hackintosh broke and I'd be grateful for your help here to better document how to use Writing Tools on macOS. Thanks!

**Reply by luke2023** [reply]
@theJayTea Hello, thanks for the reply.
Yes it did work out of the box with the Windows folder, and I think the hotkey registration in MacOS folder is buggy and not being triggered.
By the way, we have to enable terminal and python 3.12 accessibility permission in the settings app 

Lastly, thank you for the work! I'll try it later

**Reply by theJayTea** [reply]
@luke2023 there’s a new native macOS version of Writing Tools (written in Swift) created by a contributor. It’s nearly ready, and that will fix any issues with the current version on macOS and let Writing Tools work amazingly well on it :)

**Reply by luke2023** [reply]
@theJayTea Thank you all for the work, we are really looking forward to try it out and give feedback!
